pkg/drain/drain_test.go

```diff
@@ -764,3 +764,43 @@ func objBody(codec runtime.Codec, obj runtime.Object) io.ReadCloser {
 func stringBody(body string) io.ReadCloser {
 	return ioutil.NopCloser(bytes.NewReader([]byte(body)))
 }
+
+func TestBuildDeleteOptions(t *testing.T) {
+	tests := []struct {
+		description    string
+		podGracePeriod int64
+		gracePeriod    int
+		// -1 signals not set
+		expected int64
+	}{
+		{
+			description:    "library grace period seconds cannot exceed pod's",
+			podGracePeriod: 30,
+			gracePeriod:    600,
+			expected:       30,
+		},
+		{
+			description:    "library grace period negative doesn't set delete options",
+			podGracePeriod: 30,
+			gracePeriod:    -1,
+			expected:       -1,
+		},
+		{
+			description:    "library's grace period is honored if lower than pod's",
+			podGracePeriod: 30,
+			gracePeriod:    10,
+			expected:       10,
```

Comment:
@smarterclayton discussion from the thread and question: is this a case? can a lower timeout coming from let's say MCO take over the pod's timeout? otherwise, I think we can remove this option altogether from the lib and always honor the pod's deletion timeout...

Reply:
> can a lower timeout coming from let's say MCO take over the pod's timeout?

Yes.  Because [the upstream version of this library][1] is used by [the CLI][2].  And the semantics there are `min(CLI option, pod default)`.  But CLI option is a high-precedence source.  And compiled-in MCO defaults are a low-precedence source.  So we want the behaviour you have in this unit test, and we want MCO to default to `-1`.

[1]: https://github.com/kubernetes/kubernetes/blob/3317805652d145c4003f5c534c7d87fb5cd87d50/staging/src/k8s.io/kubectl/pkg/drain/drain.go#L47
[2]: https://github.com/kubernetes/kubernetes/blob/3317805652d145c4003f5c534c7d87fb5cd87d50/staging/src/k8s.io/kubectl/pkg/cmd/drain/drain.go#L191

Reply:
fixed